### ISSUE TITLE
export-name does not work with kebab/snake case

### ISSUE THREAD

**Issue by michaelw85** [Microsoft/tslint-microsoft-contrib#814]
### Bug Report

-   **`tslint-microsoft-contrib` version**: 6.0.0
-   **TSLint version**: 5.12.1
-   **TypeScript version**: 3.3.1
-   **Running TSLint via**: CLI / VSCode

#### TypeScript code being linted

```tsx
// my-component.ts || my_component.ts
export class MyComponent {
//...
}
```

with `tslint.json` configuration:

```json
{ "export-name": true }
```

#### Actual behavior
The exported module or identifier name must match the file name. Found: my-component.ts and MyComponent tslint(export-name)

#### Expected behavior
✅ Linter should pass

#### Additional request
It would also be nice to be able to accept export names where the file name matches the first part.
e.g:
```tsx
// file: src/generators/unit-test.ts
export class UnitTestGenerator {}
```

Related to #450
